rare/shared/workers/worker.py

```diff
@@ -0,0 +1,93 @@
+import sys
+from abc import abstractmethod
+from dataclasses import dataclass
+from enum import IntEnum
+from typing import Optional
+
+from PyQt5.QtCore import QRunnable, QObject, pyqtSlot, pyqtSignal
+
+
+class Worker(QRunnable):
+    """
+    Base QRunnable class.
+
+    This class provides a base for QRunnables with signals that are automatically deleted.
+
+    To use this class you have to assign the signals object of your concrete implementation
+    to the `Worker.signals` attribute and implement `Worker.run_real()`
+    """
+
+    def __init__(self):
+        sys.excepthook = sys.__excepthook__
+        super(Worker, self).__init__()
+        self.setAutoDelete(True)
+        self.__signals: Optional[QObject] = None
+
+    @property
+    def signals(self) -> QObject:
+        if self.__signals is None:
+            raise NotImplementedError
+        return self.__signals
+
+    @signals.setter
+    def signals(self, obj: QObject):
+        self.__signals = obj
+
+    @abstractmethod
+    def run_real(self):
+        pass
```

Comment:
Nit: In other `@abstractmethod`s in other files, you use ellipses (`...`) instead.

Reply:
I do not know which one I prefer either, good catch, I will decide on one style and use it throughout

Reply:
Resolved by [a3d09ae](https://github.com/Dummerle/Rare/pull/259/commits/a3d09ae288b36de8f2810409bc3bb24ee4b72db7)